src/transport/tcp/libxev.zig

```diff
@@ -0,0 +1,693 @@
+const std = @import("std");
+const xev = @import("xev");
+const Intrusive = @import("../../queue_mpsc.zig").Intrusive;
+const IOQueue = Intrusive(AsyncIOQueueNode);
+const TCP = xev.TCP;
+const Allocator = std.mem.Allocator;
+const ThreadPool = xev.ThreadPool;
+const ResetEvent = std.Thread.ResetEvent;
+
+/// SocketChannel represents a socket channel. It is used to send and receive messages.
+pub const SocketChannel = struct {
+    /// The underlying socket.
+    socket: TCP,
+
+    /// The transport that created this channel.
+    transport: *XevTransport,
+
+    /// Whether this channel is the initiator of the connection. If true, this channel is the client. If false, this channel is the server.
+    is_initiator: bool,
+
+    /// Initialize the channel with the given socket and transport.
+    pub fn init(self: *SocketChannel, socket: TCP, transport: *XevTransport, is_initiator: bool) void {
+        self.socket = socket;
+        self.transport = transport;
+        self.is_initiator = is_initiator;
+    }
+
+    /// Write sends the buffer to the other end of the channel. It blocks until the write is complete. If an error occurs, it returns the error.
+    pub fn write(self: *SocketChannel, buf: []const u8) !void {
```

Comment:
do we do a `writer` interface here? should allocator be passed expictly instead of using `transport` allocator below?

Reply:
I thought we could follow the `std.net.Stream` design which have functions returning `Reader` and `Writer`. And implementing the interface functions such as `read`, `readAll`, `readAtLeast`. These functions have no `allocator` arg. 
https://github.com/ziglang/zig/blob/master/lib/std/net.zig#L1838-L1904

The impl of the `write` and `read` function current using `allocator` internally and not involved user code, using transport's allocator may more make sense.